scene/resources/packed_scene.cpp

```diff
@@ -1589,7 +1667,94 @@ Vector<StringName> SceneState::get_node_groups(int p_idx) const {
 	return groups;
 }
 
-NodePath SceneState::get_node_path(int p_idx, bool p_for_parent) const {
+int SceneState::_find_node_by_id(int p_id) const {
+	if (id_to_node_map.size() == 0) {
+		return -1; // No node.
+	}
+
+	int map_size = id_to_node_map.size();
+	ERR_FAIL_COND_V(map_size & 1, -1);
+	const int *map = id_to_node_map.ptr();
+
+	int low = 0;
+	int high = map_size / 2;
+	int middle = 0;
+
+#ifdef DEBUG_ENABLED
+	ERR_FAIL_COND_V(low > high, -1);
+#endif
+
+	while (low <= high) {
+		middle = (low + high) / 2;
+		int pos = middle << 1;
+		int id = map[pos];
+		if (id == p_id) { //match
+			return map[pos + 1];
+		} else if (p_id < id) {
+			high = middle - 1; //search low end of array
+		} else {
+			low = middle + 1; //search high end of array
+		}
+	}
+
+	return -1;
+}
+
+NodePath SceneState::_get_path_by_index(int p_idx, bool p_expand_ids) const {
+	if (p_expand_ids) {
+		// Use cache when available.
+		if (node_paths_expanded_cache.size() == 0) {
+			node_paths_expanded_cache.resize(node_paths.size());
+		} else if (!node_paths_expanded_cache[p_idx].is_empty()) {
+			return node_paths_expanded_cache[p_idx];
+		}
+
+		Vector<StringName> ret_path;
+		Vector<StringName> path_names = node_paths[p_idx].get_names();
+		const SceneState *base_state = this; // Start by this one.
+
+		for (int i = 0; i < path_names.size(); i++) {
+			if (path_names[i].is_node_unique_id()) {
+				int32_t subindex_id = String::to_int(path_names[i].operator String().ptr() + 2);
+				int node_index = -1;
+
+				while (node_index == -1) {
+					node_index = base_state->_find_node_by_id(subindex_id);
+					if (base_state->base_scene_idx != -1) {
+						// There is inheritance at play, try with parent.
+						Ref<PackedScene> parent = base_state->variants[base_state->base_scene_idx];
+						if (parent.is_valid()) {
+							base_state = parent->get_state().ptr();
```

Comment:
This part here is pretty odd. It means if it did not find the node in the local scene, it will try with the scene that it inherits from. I am not sure if this is 100% correct, but It's the only way I can think of that works with inheritance.